ccmlib/containers/docker_client.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import socket
+from random import randrange
+import docker
+import docker.errors
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseDocker:
+    def __init__(self, name, image, port, shell_type='bash', is_clear_container=True, volumes=None,  **create_kwargs):
+        self.name = str(name)
+        self.image = image
+        self.shell_type = shell_type
+        self.container = None
+        self.container_pid = None
+        self.container_name = None
+        self.client = docker.from_env()
+        self.volumes = volumes
+
+        self.pull()
+        self.container_pid, self.container_name = self.create_container(
+            ports={"9180/tcp": port}, volumes=volumes, is_clear_container=is_clear_container, **create_kwargs)
+
+    def pull(self):
```

Comment:
why do we need this function?
to create this object, i have to already send the image to the `__init__`, so this seems useless

Reply:
Pull would be a good thing to add as a very first step, like part where we automatically download the relocatable packages.
to have a early info, if the image exist or not